### ISSUE TITLE
Cannot parse negative weights

### ISSUE THREAD

**Issue by Logan-Dang** [astashov/liftosaur#226]
I'll admit it, I can't do a pull up, and I use the assisted pullup machine. 

There are workarounds but those leave a sour taste in your mouth imo

**Comment by Logan-Dang** [comment]
So far I've tried removing !unary from liftoscript.grammar and regenerating the rules but it did not work

**Comment by astashov** [comment]
Yeah, it's kinda surprisingly tricky, so I have to admin didn't get to it yet... :)

**Comment by AnonJohn** [comment]
Also relevant to band assists and warm ups for pull ups. I've been using my body weight - assist levels as the pull up amount. Which is the amount actually being lifted. 

Further, this intersects with 1 RM and RPE calculations. Consider (using the Epley formula, 1RM = Weight * (1+ reps / 30)):

I can do 8 reps with my 180 lb body weight and a 22 lb weight on a belt. If we consider the weight lifted as 180+22, my 1 RM is 256 lbs. That means a pull up with a 76 lb weight. Plausible.

If we treat it as 22 lbs, my 1 RM is 28 lbs. This is incorrect (I have done multiple reps at that weight). This impacts programs based on 1RM %s.

Ideally I would have bodyweight as a defined variable and the ability to use negative weight for assists.

For me this comes up somewhat frequently, including with dips  and standing ab rollouts (someday I'll get one unassisted!)

Taking this to its logical end does get a little crazy. Pushups, for instance, are a bodyweight exercise. For most people, you are pushing ~70% of your body weight if your feet are on the floor. There is variance depending on how your bodyweight is distributed, and more if you elevate your feet etc). In a perfect world I would imagine configuring body weight exercises with a changeable % of body weight, but that is probably more trouble than it is worth.

**Comment by richbeales** [comment]
@astashov this can probably be closed due to https://github.com/astashov/liftosaur/commit/55ca42900547dd21741d1f9b844133bbea2b7754
